presqt/api_v1/views/resource.py

```diff
@@ -83,9 +85,109 @@ def get(self, request, target_name):
         # Fetch the target's resources
         try:
             resources = func(token)
-        except Exception as e:
+        except PresQTResponseException as e:
             # Catch any errors that happen within the target fetch
-            return Response(data={'error': e.__str__()}, status=status.HTTP_400_BAD_REQUEST)
+            return Response(data={'error': e.data}, status=e.status_code)
 
         serializer = ResourcesSerializer(instance=resources, many=True)
         return Response(serializer.data)
+
+
+class Resource(APIView):
```

Comment:
If you prefer, you could probably rename ResourceCollection to Resources.
Then you'd have the singular (Resource) and plural (Resources).

Just a thought.